repository/main.tf

```diff
@@ -0,0 +1,89 @@
+resource "aws_ecr_repository" "ecr-repo" {
+  name = "${var.repository_name}"
+}
+
+resource "aws_iam_policy" "ecr-repo-push" {
+  name        = "ecr-${var.repository_name}-push"
+  path        = "/"
+  description = "Push access to the ${var.repository_description} repository"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:GetAuthorizationToken",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:GetRepositoryPolicy",
+        "ecr:CompleteLayerUpload",
+        "ecr:InitiateLayerUpload",
+        "ecr:PutImage",
+        "ecr:UploadLayerPart"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_ecr_repository.ecr-repo.arn}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "ecr-repo-pull" {
+  name        = "ecr-${var.repository_name}-pull"
+  path        = "/"
+  description = "Pull access to the ${var.repository_description} repository"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:GetAuthorizationToken",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:GetRepositoryPolicy"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_ecr_repository.ecr-repo.arn}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_ecr_lifecycle_policy" "ecr-repo-policy" {
```

Comment:
Can we make this optional? It could be disabled by setting `expire_after` to 0 or something like that.

Reply:
@iuriaranda I would prefer automatic cleanup of these *untagged* images. Don't forget that:

1. an ECR repository has a limit of 1000 images
1. [you pay](https://aws.amazon.com/ecr/pricing/) for stored images

Reply:
I understand what you mean, although it wouldn't hurt to make it optional tbh. You can still set the default value to 30 days, so it still cleans up these images by default, and only disable that behavior if you explicitly say so.